app/Http/Controllers/UserController.php

```diff
@@ -245,6 +251,38 @@ public function postProfilePasswordEdit(Request $request)
         return redirect()->back()->with('error', 'Current Password does not match!');
     }
 
+    public function postProfileRepairDirectory(Request $request)
+    {
+        $rules = [
+            'role' => 'required|digits_between:' . Role::REPAIR_DIRECTORY_SUPERADMIN . ',' . Role::REPAIR_DIRECTORY_EDITOR
+        ];
+
+        $validator = Validator::make($request->all(), $rules);
+
+        if ($validator->fails()) {
+            return redirect()->back()->withErrors($validator)->withInput();
+        }
+
+        $id = $request->input('id');
+        $role = intval($request->input('role'));
+        $user = User::find($id);
+
+        // Check that we are allowed to change the role, based on our own role.
+        try {
+            $this->authorize('canChangeRepairDirRole', Auth::user(), $user, $role);
```

Comment:
I think we might not need this try/catch, as I believe $this->authorize will trigger some Laravel error handling to redirect with a 403 Forbidden if the authorization fails.  

If we just catch the exception and log it, wouldn't we then continue executing the rest of the code and still end up updating the role?

Reply:
Thanks, you're right - I had this try in as debug but it can now go.